lib/connectionConfig/connectionConfig.ts

```diff
@@ -112,25 +111,38 @@ export namespace ConnectionConfig {
    */
   export function validate(config: ConnectionConfig, options?: ConnectionConfigOptions): void {
     if (!options) options = {};
-    if (!config || (config && typeof config !== "object")) {
-      throw new Error("'config' is a required parameter and must be of type: 'object'.");
+
+    if (!config) {
+      throw new TypeError("Missing configuration");
+    }
+
+    if (!config.endpoint) {
+      throw new TypeError("Missing 'endpoint' in configuration");
     }
-    if (!config.endpoint || (config.endpoint && typeof config.endpoint !== "string")) {
-      throw new Error("'endpoint' is a required property of ConnectionConfig.");
+    config.endpoint = String(config.endpoint);
+
+    if (config.entityPath) {
+      config.entityPath = String(config.entityPath);
     }
-    if (config.entityPath && typeof config.entityPath !== "string") {
-      throw new Error("'entityPath' must be of type 'string'.");
+
+    if (config.host) {
+      config.host = String(config.host);
     }
-    if (options.isEntityPathRequired && !config.entityPath) {
-      throw new Error("'entityPath' is a required property of ConnectionConfig.");
+
+    if (options.isEntityPathRequired) {
+      if (!config.entityPath) {
```

Comment:
The `config.entityPath` is being checked twice. We can refactor as follows:
```
if (config.entityPath) {
   config.entityPath = String(config.entityPath)
} else if (options.isEntityPathRequired) {
  // throw the error here
}
```

Reply:
Also we can refactor like this:
```typescipt
if (options.isEntityPathRequired && !config.entityPath) {
       // throw error
    } 
  config.entityPath = String(config.entityPath);
```